src/eth_rpc/handlers/eth_get_block_by_number.rs

```diff
@@ -49,29 +43,28 @@ pub async fn handle(
     let query_ranges_for_headers =
         optimize_query_for_single_block_request(from_blocks_for_headers, rpc_handler.max_block_gap);
 
-
     // execute skar query
     let res_block_txns =
-        execute_query_for_block_txns(rpc_handler.skar_client.clone(), query_ranges_for_txns).await;
-    let res_block_headers =
-        execute_query_for_block_headers(rpc_handler.skar_client.clone(), query_ranges_for_headers).await;
+        execute_query_for_block_txns(rpc_handler.query_handler.clone(), query_ranges_for_txns)
```

Comment:
What is the reason we don't use skar for this query?

Reply:
Query handler uses a skar client internally to execute the query. It just has some more functionality on top to convert the types